src/cargo-fmt/main.rs

```diff
@@ -249,10 +249,16 @@ fn get_targets(strategy: &CargoFmtStrategy) -> Result<HashSet<Target>, io::Error
 
 fn get_targets_root_only(targets: &mut HashSet<Target>) -> Result<(), io::Error> {
     let metadata = get_cargo_metadata(None)?;
+    let current_dir = env::current_dir()?;
+    let current_dir_manifest = current_dir.join("Cargo.toml");
+    let workspace_root_path = PathBuf::from(&metadata.workspace_root);
+    let in_workspace_root = workspace_root_path == current_dir;
```

Comment:
You should probably canonicalise the paths before comparing them (although tbh, I don't know how equality for paths work so it might not be necessary).